### ISSUE TITLE
Accessing .NET Core 2.1 types, features in class library (e.g. Type.IsByRefLike)

### ISSUE THREAD

**Issue by kierenj** [dotnet/coreclr#21161]
First, please forgive me.  I'm almost certain I don't have the absolute-best repo here, but I'll be honest, I don't know what lies with CoreCLR, what's with CoreFX, etc etc.

I'm upgrading my little framework to .NET Core 2.1.  I'm guessing it's bad timing since NuGet UI is giving me 2.2.0, despite it not being announced or on any site, but that's beside the point.

My sample apps are fine with TFM `netcoreapp2.1`.  But I couldn't see any guidance anywhere on if/how to upgrade class libraries, which this project mostly is.  I have unit tests which pass (because they target `netstandard2.0`) but fail at runtime, because `Type.IsByRefLike` exists on `Type`, it seems, only in `netcoreapp2.1`.  Accessing the property simply won't compile on `netstandard2.0`.

My questions are:

1. What should a class library do if it wants to use `netcoreapp2.1` features, like `Type.IsByRefLike`?
2. What is the guidance for class libraries which are built on/around .NET Core 2.1 libraries from Microsoft?  Keep to `netstandard2.0`?  Do whatever it is you suggest for Q1?

**Comment by akoeplinger** [comment]
`Type.IsByRefLike` is scheduled for the upcoming .NET Standard 2.1: https://github.com/dotnet/standard/blob/master/docs/versions/netstandard2.1_diff.md

You can either wait until that is available or target `netcoreapp2.1` with your class library (which makes it only compatible with .NET Core of course).

**Comment by kierenj** [comment]
Ok, perfect, thanks.  I'm sure I've seen in "how to upgrade to .NET Core 2.0" guides which TFM to use in each case, but I really couldn't find any docs for .NET Core 2.1.

It felt very wrong to put `netcoreapp2.1` in class libraries, but if there's no other way, that's what I'll crack on with, thanks.

Is there anything that could/should be shared in docs around this?

**Comment by akoeplinger** [comment]
The doc at https://docs.microsoft.com/en-us/dotnet/standard/frameworks does mention that "An app or *__library__* can also target a specific .NET implementation to gain access to implementation-specific APIs." but that can maybe improved to make clear it applies to .NET Core libraries too.

I assume you got sidetracked by the word `app` in netcoreapp, the historical background is that the `netcore` TFM was already taken by Windows Store apps so a new one had to be used.

**Comment by kierenj** [comment]
Yes exactly, it said app - and I wasn't building an app.  Understood thanks.  It was a bit frustrating that the only "how to upgrade to .NET Core 2.1" guide I could find was talking about ASP.NET Core apps.  I definitely remember something more in-depth for 1.0 to 2.0, which is understandable.  And in the end someone kind & helpful got me there :+1:, so maybe it's not a big issue, thanks.
